### PR TITLE
test: fix mocked commands

### LINKED ISSUE #1484: Ensure the mocked commands match the actual output of the actual commands
#### Overview

We need to make sure that all the output of the mocked commands in [`utits.rs`](https://github.com/starship/starship/blob/master/src/utils.rs#L41-L145) match the output of the actual commands, including any trailing whitespace.

#### Context

> Not sure what to think about this 🤔  Thing is, most of the other stdout/stderr outputs also lack trailing newline so we should probably fix all of them inside separate PR?

Yep, this is something we need to look at. The mocks should match the output of the commands exactly, including trailing whitespace, this is something that we haven't been doing very well so far and this is the second regression that this has caused the other being #1465. I'll create an issue to fix this for all the existing modules.

_Originally posted by @andytom in https://github.com/starship/starship/pull/1479_

### PR DESCRIPTION
**Pull request by vladimyr**
#### Description

Ensure that output of mocked commands faithfully replicates output of actual commands including any trailing whitespace.

##### Todo list:

- [x] `crystal --version`
- [x] `dummy_command`
- [x] `elixir --version`
- [x] `elm --version`
- [x] `go version`
- [x] `helm version --short --client`
- [x] `julia --version`
- [x] `nim --version`
- [x] `node --version`
- [x] `ocaml -vnum`
- [x] `esy ocaml -vnum`
- [x] `php -nr 'echo PHP_MAJOR_VERSION.".".PHP_MINOR_VERSION.".".PHP_RELEASE_VERSION;'`
- [x] `purs --version`
- [x] `python --version` (python2)
- [x] `python3 --version`
- [x] `ruby -v`
- [ ] ~`stack --no-install-ghc --lock-file read-only ghc -- --numeric-version`~ (removed in #1418)
- [x] `zig version`
- [x] `cmake --version`
- [x] `terraform version`
- [x] `erl -noshell -eval...` (erlang)

#### Motivation and Context

Closes #1484 

#### How Has This Been Tested?

- [x] I have tested using **MacOS**
- [ ] I have tested using **Linux**
- [ ] I have tested using **Windows**

#### Checklist:

- [ ] I have updated the documentation accordingly.
- [x] I have updated the tests accordingly.

### CODE REVIEW

File: src/utils.rs [hunk 1]
```diff
@@ -106,15 +111,15 @@ active boot switches: -d:release\n",
             stderr: String::default(),
         }),
         "python --version" => Some(CommandOutput {
-            stdout: String::from("Python 2.7.17"),
-            stderr: String::default(),
+            stdout: String::default(),
+            stderr: String::from("Python 2.7.17\n"),
```
Comment (vladimyr) [inline src/utils.rs:115]: ⚠️ Python 2 outputs version on stderr!

File: src/utils.rs [hunk 2]
```diff
@@ -95,26 +100,26 @@ active boot switches: -d:release\n",
...
-        "php -nr echo PHP_MAJOR_VERSION.'.'.PHP_MINOR_VERSION.'.'.PHP_RELEASE_VERSION;" => {
+        "php -nr 'echo PHP_MAJOR_VERSION.\".\".PHP_MINOR_VERSION.\".\".PHP_RELEASE_VERSION;'" => {
             Some(CommandOutput {
                 stdout: String::from("7.3.8"),
                 stderr: String::default(),
             })
         }
         "purs --version" => Some(CommandOutput {
-            stdout: String::from("0.13.5"),
+            stdout: String::from("0.13.5\n"),
```
Comment (vladimyr) [inline src/utils.rs:110]: ⚠️ PureScript installed using Homebrew reports following:
```
starship on  fix-mocked-commands [⇕$?] is 📦 v0.44.0 via 🦀 v1.44.1
❯ /usr/local/opt/purescript/bin/purs --version | cat -A --plain
0.13.8•[development•build;•commit:•UNKNOWN]␊
```

File: src/utils.rs (hunk 1 above)
Comment (andytom) [reply src/utils.rs:116]: Good catch!

File: src/utils.rs [hunk 3]
```diff
@@ -40,28 +40,34 @@ pub fn exec_cmd(cmd: &str, args: &[&str]) -> Option<CommandOutput> {
...
-            stdout: String::from("0.19.1"),
+            stdout: String::from("0.19.1\n"),
             stderr: String::default(),
         }),
         "go version" => Some(CommandOutput {
-            stdout: String::from("go version go1.12.1 linux/amd64"),
+            stdout: String::from("go version go1.12.1 linux/amd64\n"),
             stderr: String::default(),
         }),
         "helm version --short --client" => Some(CommandOutput {
```
Comment (andytom) [inline src/utils.rs:73]: Github doesn't let me add this as a suggestion on the line below but the output from helm includes a new line
```
❯ helm version --short --client | cat -A
v3.2.4+g0ad800e$
```

File: src/utils.rs [hunk 4]
```diff
@@ -140,8 +146,8 @@ CMake suite maintained and supported by Kitware (kitware.com/cmake).\n",
             stdout: String::from("Terraform v0.12.14"),
```
Comment (andytom) [inline src/utils.rs:146]: Terraform version includes a new line,
```
❯ terraform version | cat -A
Terraform v0.12.28$
```

```suggestion
            stdout: String::from("Terraform v0.12.14\n"),
```

File: src/utils.rs [hunk 5]
```diff
@@ -95,26 +101,26 @@ active boot switches: -d:release\n",
...
-            stdout: String::from("Python 3.8.0"),
+            stdout: String::from("Python 3.8.0\n"),
             stderr: String::default(),
         }),
         "ruby -v" => Some(CommandOutput {
-            stdout: String::from("ruby 2.5.1p57 (2018-03-29 revision 63029) [x86_64-linux-gnu]"),
+            stdout: String::from("ruby 2.5.1p57 (2018-03-29 revision 63029) [x86_64-linux-gnu]\n"),
             stderr: String::default(),
         }),
         "stack --no-install-ghc --lock-file read-only ghc -- --numeric-version" => {
```
Comment (andytom) [inline src/utils.rs:126]: We removed the Haskell module in #1418, so we can remove this mock.

**Comment by andytom** [comment]
@vladimyr, sorry about the comment spam but I've tried to add some suggestions to fill in the gaps.

**Comment by vladimyr** [comment]
> @vladimyr, sorry about the comment spam but I've tried to add some suggestions to fill in the gaps. 

No worries, I would subscribe to such _spam_ any day of the week 😀, thank you! 👍

File: src/utils.rs (hunk 2 above)
Comment (vladimyr) [reply src/utils.rs:111]: The fix is on its way: https://github.com/Homebrew/homebrew-core/pull/58216

**Pull request merged**